src/main/java/catan/services/util/play/PlayUtil.java

```diff
@@ -0,0 +1,135 @@
+    private Logger log = LoggerFactory.getLogger(PlayUtil.class);
+
+    private EndTurnUtil endTurnUtil;
+
+    private static final Gson GSON = new Gson();
+
+    public List<List<String>> getInitialBuildingsSetFromJson(String json) {
+        return GSON.fromJson(json, new TypeToken<List<List<String>>>() {
+        }.getType());
+    }
+
+    public AllAvailableActionsDetails getAllAvailableActions(String json) {
+        return GSON.fromJson(json, AllAvailableActionsDetails.class);
+    }
+
+    public void updateNextMoveOrder(GameBean game) throws GameException {
+        Integer nextMoveNumber;
+        switch (game.getStage()) {
+            case PREPARATION:
+                List<List<String>> initialBuildingsSet = getInitialBuildingsSetFromJson(game.getInitialBuildingsSet());
+                nextMoveNumber = endTurnUtil.endTurnImplInPreparationStage(game, initialBuildingsSet);
+                break;
+            case MAIN:
+                nextMoveNumber = endTurnUtil.endTurnImplInMainStage(game);
+                break;
+            default:
+                log.debug("Cannot recognize current game stage: {}", game.getStage());
+                throw new GameException(ERROR_CODE_ERROR);
+        }
+
+        log.debug("Next move order in {} stage is changing from {} to {}", game.getStage(), game.getCurrentMove(), nextMoveNumber);
+        game.setCurrentMove(nextMoveNumber);
+    }
+
+    public void updateCurrentCycleBuildingNumber(GameBean game) {
+        if (game.getStage().equals(GameStage.PREPARATION)) {
+            List<List<String>> initialBuildingsSet = getInitialBuildingsSetFromJson(game.getInitialBuildingsSet());
+            if (game.getCurrentCycleBuildingNumber() > initialBuildingsSet.get(game.getPreparationCycle() - 1).size()) {
+                game.setCurrentCycleBuildingNumber(1);
+            } else {
+                game.setCurrentCycleBuildingNumber(game.getCurrentCycleBuildingNumber() + 1);
+            }
+        }
+    }
+    public void updateAvailableUserActions(GameBean game) throws GameException {
+        boolean isMandatory = false;
+        List<String> actionsList = new ArrayList<String>();
+        List<GameUserActions> actionCode = new ArrayList<GameUserActions>();
+        switch (game.getStage()) {
+            case PREPARATION:
+                List<List<String>> initialBuildingsSet = getInitialBuildingsSetFromJson(game.getInitialBuildingsSet());
+                for (GameUserBean gameUser : game.getGameUsers()) {
+                    if (gameUser.getMoveOrder() == game.getCurrentMove()) {
+                        actionCode = getActionCodeInPreparationStage(game, initialBuildingsSet);
+                    }
+                    if (!actionCode.isEmpty()) {
+                        actionsList.add("{\"code\": \"" + actionCode.get(0).toString() + "\", \"params\": {}}");
+                        gameUser.setAvailableActions("{\"list\": [" + actionsList.get(0) + "], \"isMandatory\": true}");
```

Comment:
in praparation step we have only one mandatory action, always. so there is no nesessary to iterate by all values here